### PR TITLE
Disable Http status line parsing test cases on CurlHandler

### PR DESCRIPTION
**Pull request by caesar1995**
Closes: #26980

### CODE REVIEW

File: src/System.Net.Http/tests/FunctionalTests/HttpProtocolTests.cs [hunk 1]
```diff
@@ -435,6 +427,19 @@ public static IEnumerable<string> GetInvalidStatusLine()
                 yield return "HTTP/1.12 200 OK";
                 yield return "HTTP/12.1 200 OK";
             }
+
+            // Skip these test cases on CurlHandler since the behavior is different.
+            if (PlatformDetection.IsWindows)
```
Comment (caesar1995) [inline src/System.Net.Http/tests/FunctionalTests/HttpProtocolTests.cs]: I don't realize any general Unix PlatformDetection (there is specific detection, like OSX), so I take the reverse, which is only run on Windows.

Comment (davidsh) [reply src/System.Net.Http/tests/FunctionalTests/HttpProtocolTests.cs]: >I don't realize any general Unix PlatformDetection (there is specific detection, like OSX)

There are other classes/properties that can be used for Unix detection (which includes Mac OSX and Linux).

Comment (caesar1995) [reply src/System.Net.Http/tests/FunctionalTests/HttpProtocolTests.cs]: The `[PlatformSpecific(TestPlatforms.AnyUnix)]` can only be applied to test method, not for static test data.

**Comment by karelz** [comment]
Thanks for fast turnaround!

**Pull request merged**